### ISSUE TITLE
[Dialog] Improve accessibility of the dialog title

### ISSUE THREAD

**Issue by anna-lach** [sl-design-system/components#2846]
### Provide a general summary of the issue here

Right now, the title in the dialog is not a heading, but it should be.

We need to use `h2` for the title in the dialog instead of span "title" (`h1` can be already used somewhere on the page, so it's better to use `h2` in this case).

### 👤 Your name

Anna Sobczak

### 🧢 Your Product/Team

SLDS
